src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/DefaultRazorProjectChangePublisherTest.cs

```diff
@@ -51,8 +52,101 @@ public class DefaultRazorProjectChangePublisherTest : LanguageServerTestBase
     {
         private readonly RazorLogger RazorLogger = Mock.Of<RazorLogger>();
 
+        public DefaultRazorProjectChangePublisherTest()
+        {
+            ProjectSnapshotManager = CreateProjectSnapshotManager(allowNotifyListeners: true);
+        }
+
+        private ProjectSnapshotManagerBase ProjectSnapshotManager { get; }
+
         private ProjectConfigurationFilePathStore ProjectConfigurationFilePathStore { get; } = new DefaultProjectConfigurationFilePathStore();
 
+        [Fact]
+        public void ProjectManager_Changed_NotActive_Noops()
+        {
+            // Arrange
+            var attemptedToSerialize = false;
+            var hostProject = new HostProject("/path/to/project.csproj", RazorConfiguration.Default, rootNamespace: "TestRootNamespace");
+            var hostDocument = new HostDocument("/path/to/file.razor", "file.razor");
+            ProjectSnapshotManager.ProjectAdded(hostProject);
+            var publisher = new TestDefaultRazorProjectChangePublisher(
+                ProjectConfigurationFilePathStore,
+                RazorLogger,
+                onSerializeToFile: (snapshot, configurationFilePath) => attemptedToSerialize = true)
+            {
+                EnqueueDelay = 10,
+            };
+            publisher.Initialize(ProjectSnapshotManager);
+
+            // Act
+            ProjectSnapshotManager.DocumentAdded(hostProject, hostDocument, new EmptyTextLoader(hostDocument.FilePath));
```

Comment:
It seems like there should have been an await needed somewhere in here to make sure there was time for the `onSerializeToFile` callback to be called?

Reply:
In this specific test we validate that nothing was called. Basically that it no-ops and therefore no async work required